test/set-context-property

```diff
@@ -3,37 +3,58 @@
 import sys
 import dbus
 
-if len(sys.argv) < 4:
-	print("Usage: set-context-property <context> <name> <value>")
-	sys.exit(1)
-
 bus = dbus.SystemBus()
 
-manager = dbus.Interface(bus.get_object('org.ofono', '/'),
-						'org.ofono.Manager')
+manager = dbus.Interface(bus.get_object('org.ofono', '/'), 'org.ofono.Manager')
+
+if (len(sys.argv) == 3):
+	modems = manager.GetModems()
+	modem = modems[0][0]
+	context_idx = 0
+	property = sys.argv[1]
+	value = sys.argv[2]
+elif (len(sys.argv) == 4):
+	modems = manager.GetModems()
+	modem = modems[0][0]
+	context_idx = int(sys.argv[1]) - 1
+	property = sys.argv[2]
+	value = sys.argv[3]
+elif (len(sys.argv) == 5):
+	modem = sys.argv[1]
+	context_idx = int(sys.argv[2]) - 1
+	property = sys.argv[3]
+	value = sys.argv[4]
+else:
+	print("Usage: set-context-property [modem] [context_number] <name> <value>")
+	sys.exit(1)
 
-modems = manager.GetModems()
+modemapi = dbus.Interface(bus.get_object('org.ofono', modem), 'org.ofono.Modem')
+properties = modemapi.GetProperties()
 
-for path, properties in modems:
-	if "org.ofono.ConnectionManager" not in properties["Interfaces"]:
-		continue
+if "org.ofono.ConnectionManager" not in properties["Interfaces"]:
+	print("org.ofono.ConnectionManager not found")
+	exit(2)
 
-	connman = dbus.Interface(bus.get_object('org.ofono', path),
+connman = dbus.Interface(bus.get_object('org.ofono', modem),
 					'org.ofono.ConnectionManager')
 
-	contexts = connman.GetContexts()
+contexts = connman.GetContexts()
+
+if (len(contexts) == 0):
+	print("No context available")
+	sys.exit(1)
+
+path = contexts[context_idx][0]
 
-	if (len(contexts) == 0):
-		print("No context available")
-		sys.exit(1)
+context = dbus.Interface(bus.get_object('org.ofono', path),
+				'org.ofono.ConnectionContext')
 
-	path = contexts[int(sys.argv[1])][0]
-	context = dbus.Interface(bus.get_object('org.ofono', path),
-					'org.ofono.ConnectionContext')
+if property == "Preferred":
+	value = dbus.Boolean(value, variant_level=1)
```

Comment:
Trying

./set-context-property /ril_0 2 Preferred 0
or
./set-context-property /ril_0 2 Preferred false

still does not work and Preferred does not change if it was set to true. I think that the problem is that this is a direct conversion from string to boolean, which seems to be true always.

value = dbus.Boolean(int(value))

works if value is 0/1 for instance.